### PR TITLE
reduce memory used by TemplateInstance instances

### PR DESCRIPTION
**Pull request by ghost**
inherited member `ScopeDsymbol.endlinnum` was never used. Just like in #11365, I observe a reduction of 2Mb of the max resident set size when compiling dparse, as the TemplateInstance instances take a small memory block in less.

### CODE REVIEW

File: src/dmd/dsymbolsem.d [hunk 1]
```diff
@@ -1551,13 +1551,6 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
 
         if (dsym.type.toBasetype().ty == Terror)
             dsym.errors = true;
-
-        if(sc.scopesym && !sc.scopesym.isAggregateDeclaration())
-        {
-            for (ScopeDsymbol sym = sc.scopesym; sym && dsym.endlinnum == 0;
-                 sym = sym.parent ? sym.parent.isScopeDsymbol() : null)
-                dsym.endlinnum = sym.endlinnum;
```
Comment (rainers) [inline src/dmd/dsymbolsem.d:1559]: This is valid usage of the member necessary for debug information.

**Review by rainers — CHANGES_REQUESTED** [review]
endlinnum should not be removed

File: src/dmd/dsymbolsem.d (hunk 1 above)
Comment (NilsLankila) [reply src/dmd/dsymbolsem.d:1559]: can you show me where this is used, I've  browsed files and it doedsn't seem to be used. by cv.d and alikes.

Comment (rainers) [reply src/dmd/dsymbolsem.d:1559]: It's here: https://github.com/dlang/dmd/blob/master/src/dmd/tocsym.d#L319

Comment (NilsLankila) [reply src/dmd/dsymbolsem.d:1559]: tocsym was not modified by this PR, looks like it's another linnum maybe

Comment (rainers) [reply src/dmd/dsymbolsem.d:1559]: dsym is of type VarDeclaration, not DScopeSymbol.

Comment (NilsLankila) [reply src/dmd/dsymbolsem.d:1559]: I see, it's used but indirectly, after copy.

Comment (NilsLankila) [reply src/dmd/dsymbolsem.d:1559]: Maybe that a an offset from loc.linnum could be stored in a ushort instead of the full value

**Pull request closed without merging**